src/coreclr/gc/gc.cpp

```diff
@@ -10417,30 +10428,40 @@ void gc_heap::return_free_region (heap_segment* region)
     }
 }
 
+// USE_REGIONS TODO: SOH should be able to get a large region and split it up into basic regions
+// if needed.
 // USE_REGIONS TODO: In Server GC we should allow to get a free region from another heap.
 heap_segment* gc_heap::get_free_region (int gen_number)
 {
     heap_segment* region = 0;
-    if (free_regions)
+
+    // TODO: the update to committed_in_free is incorrect - we'd need synchorization 'cause a thread
+    // could be getting a small and a large region at the same time. This is only used for recording.
```

Comment:
This sounds like a single thread could be getting a small and a large region at the same time. What you meant is probably that one thread could be getting a small region (under the SOH lock), and another thread could be getting a large region (under the LOH lock).

Reply:
yes that's what I meant...will change,thanks!